static/app/views/settings/organizationIntegrations/integrationServerlessRow.tsx

```diff
@@ -15,137 +14,131 @@ import type {
+        action,
+        organization,
+      });
+    },
+    [integration.provider.key, organization]
+  );
+
+  const handleUpdate = useCallback(async () => {
     const data = {
-      action,
+      action: 'updateVersion',
       target: serverlessFunction.name,
     };
     try {
+      setIsSubmitting(true);
+      // don't know the latest version but at least optimistically remove the update button
+      onUpdate({outOfDate: false});
       addLoadingMessage();
-      this.setState({submitting: true});
-      // optimistically update enable state
-      this.props.onUpdateFunction({enabled: !this.enabled});
-      this.recordAction(action);
-      const resp = await this.props.api.requestPromise(this.endpoint, {
+      recordAction('updateVersion');
+      const resp = await api.requestPromise(endpoint, {
         method: 'POST',
         data,
       });
       // update remaining after response
-      this.props.onUpdateFunction(resp);
+      onUpdate(resp);
       addSuccessMessage(t('Success'));
     } catch (err) {
       // restore original on failure
-      this.props.onUpdateFunction(serverlessFunction);
+      onUpdate(serverlessFunction);
       addErrorMessage(err.responseJSON?.detail ?? t('Error occurred'));
     }
-    this.setState({submitting: false});
-  };
-  updateVersion = async () => {
-    const {serverlessFunction} = this.props;
+    setIsSubmitting(false);
+  }, [api, endpoint, onUpdate, recordAction, serverlessFunction]);
+
+  const handleToggle = useCallback(async () => {
+    const action = serverlessFunction.enabled ? 'disable' : 'enable';
     const data = {
-      action: 'updateVersion',
+      action,
       target: serverlessFunction.name,
     };
     try {
-      this.setState({submitting: true});
-      // don't know the latest version but at least optimistically remove the update button
-      this.props.onUpdateFunction({outOfDate: false});
       addLoadingMessage();
-      this.recordAction('updateVersion');
-      const resp = await this.props.api.requestPromise(this.endpoint, {
+      setIsSubmitting(true);
+      // optimistically update enable state
+      onUpdate({enabled: !serverlessFunction.enabled});
+      recordAction(action);
+      const resp = await api.requestPromise(endpoint, {
         method: 'POST',
         data,
       });
       // update remaining after response
-      this.props.onUpdateFunction(resp);
+      onUpdate(resp);
       addSuccessMessage(t('Success'));
     } catch (err) {
       // restore original on failure
-      this.props.onUpdateFunction(serverlessFunction);
+      onUpdate(serverlessFunction);
       addErrorMessage(err.responseJSON?.detail ?? t('Error occurred'));
     }
-    this.setState({submitting: false});
-  };
-  renderLayerStatus() {
-    const {serverlessFunction} = this.props;
+    setIsSubmitting(false);
+  }, [api, endpoint, onUpdate, recordAction, serverlessFunction]);
+
+  const layerStatus = useMemo(() => {
```

Comment:
That makes sense! I didn't notice that with the `onUpdate` dep. I'll probably leave it as is here though just because I like that it isolates all the logic for determining what this layer status can be in some.